### DISCUSSION TITLE
\n isn't triggering Enter even when doing wezterm cli send-text --no-paste

### DISCUSSION THREAD

**Discussion by weiying-chen** [wezterm/wezterm#4950] [Q&A]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter

### WezTerm version

20240127-113634-bbcac864

### Ask your question!

When I run this command:

```
wezterm cli send-text --no-paste "echo Test\n"
```

I get this:

```
➜  wezterm cli send-text --no-paste "echo Test\n"
echo test\n%
➜  echo test\n
```

The `\n` doesn't trigger an Enter (carriage return) as mentioned [here](https://github.com/wez/wezterm/issues/3891#issuecomment-1603615149).

Notes:
- I'm using zsh on Ubuntu.
- I also tried `\r`, but I get the same result.

**Comment by weiying-chen** [comment]
This works:

```
echo ":open ${fpath}\r" | wezterm cli send-text --no-paste
```

**Reply by psnelgrove-r7** [reply]
Today I ran into the same thing. My plan was to combine `watchexec` with sending a `:reload\r` message to Helix to reload the editor on a file that was being generated.

I'm not sure how to debug something like this (strace or similar I'm sure), but it's either echo sending the bytes differently to stdout or wezterm is interpreting them differently from stdin. 

I'm at the point of blowing my budget for tinkering for a perceived optimization in my workflow, but if I can debug a little more later I will.

**Reply by psnelgrove-r7** [reply]
So the [implementation](https://github.com/wez/wezterm/blob/4050072da21cc3106d0985281d75978c07e22abc/wezterm/src/cli/send_text.rs#L27) of the command is dead simple.

```rust
let data = match self.text {
    Some(text) => text,
    None => {
        let mut text = String::new();
        std::io::stdin()
            .read_to_string(&mut text)
            .context("reading stdin")?;
        text
    }
};
```
So I'd wager then that it's something with how the `clap` parser handles the string.

**Reply by psnelgrove-r7** [reply]
Alright, really need to get back to work but I think this is close to the answer. When `clap` loads up the string the escaped characters are being interpreted rather than a raw string. However the stdin stream is a sequence of bytes.

https://users.rust-lang.org/t/parsing-whitespace-vs-their-escape-sequences/88771
